### PR TITLE
feat(theming): introduce focus-visible to ThemeProvider

### PR DESCRIPTION
**Pull request by austin94**
## Description

This PR adds the `useFocusVisible()` hook to our `ThemeProvider`. This allows all Garden elements to use its data attributes for `:focus-visible` styling.

This implementation:
- Uses an immediate `<div>` element to apply the focus-visible boundary
  - This avoids leaking data-attributes outside of our area of concern
  - This also allows nested `ThemeProvider`s to work independently
- Can accept a `focusVisibleRef` prop which allows consumers to not have the `<div>` rendered
  - We use this in our `react-testing` utilities to avoid updating all our tests
- Can customize the `relativeDocument` value through our `theme` object

## Details

I have also removed some of our focused tests as they are only testing the functionality of the `useFocusVisible()` hook.

## Checklist

- [ ] ~:ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)~
- [ ] ~:nail_care: view component styling is based on a Garden CSS
      component~
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] ~:guardsman: includes new unit tests~
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by austin94** [comment]
@hzhu I think the relative link that `Anchor` was using is invalid in the `next` branch as the examples have changed names. I'll look into this.

File: packages/buttons/examples/button.md [hunk 1]
```diff
@@ -57,14 +57,19 @@ button props.
...
+to apply focused styling. This allows us to mimic the functionality of the
+[:focus-visible](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible)
+pseudo-class.
+
 ```jsx
 <Grid>
   <Row>
     <Col md>
-      <Button focused>Focused Default</Button>
+      <Button data-garden-focus-visible>Focused Default</Button>
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/buttons/examples/button.md]: To be clear – CSS state (hover, focused, active, etc) should be eliminated. Too much visual clutter without documentation benefit. aka, we don't ever want people tweaking these in production.

[the remark this answers is not in the document]
Comment (austin94) [reply packages/buttons/examples/button.md]: 👍 Agreed. Just wanted to get the examples to parity with this PR. We can remove these with a global task to make sure we remove all references.

File: packages/buttons/examples/button.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ button props.
 ### States
 
 Garden elements use `data` attributes provided by our `<ThemeProvider>`
```
Comment (jzempel) [inline packages/buttons/examples/button.md]: ~our~

**Comment by jzempel** [comment]
@austin94 after you merge, I'll plan to update #507 accordingly.

**Pull request merged**